src/components/panningViews/panToDismissView.js

```diff
@@ -0,0 +1,301 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Animated} from 'react-native';
+import {BaseComponent} from '../../commons';
+import {Constants} from '../../helpers';
+import asPanViewConsumer from './asPanViewConsumer';
+import PanningProvider from './panningProvider';
+
+const DEFAULT_SPEED = 20;
+const DEFAULT_BOUNCINESS = 6;
+const DEFAULT_DISMISS_ANIMATION_DURATION = 280;
+
+/**
+ * @description: PanToDismissView component created to making listening to swipe and drag events easier,
+ * Has to be used as a child of a PanningProvider that also has a PanListenerView
+ */
+class PanToDismissView extends BaseComponent {
+  static displayName = 'PanToDismissView';
+  static propTypes = {
+    /**
+     * The directions of the allowed pan (default allows all directions)
+     * Types: UP, DOWN, LEFT and RIGHT (using PanningProvider.Directions.###)
+     */
+    directions: PropTypes.arrayOf(PropTypes.oneOf(Object.values(PanningProvider.Directions))),
+    /**
+     * onDismiss callback
+     */
+    onDismiss: PropTypes.func,
+    /**
+     * The animation speed (default is 20)
+     */
+    animationSpeed: PropTypes.number,
+    /**
+     * The animation bounciness (default is 6)
+     */
+    animationBounciness: PropTypes.number,
+    /**
+     * The dismiss animation duration (default is 280)
+     */
+    dismissAnimationDuration: PropTypes.number,
+  };
+
+  static defaultProps = {
+    directions: [
+      PanningProvider.Directions.UP,
+      PanningProvider.Directions.DOWN,
+      PanningProvider.Directions.LEFT,
+      PanningProvider.Directions.RIGHT,
+    ],
+    animationSpeed: DEFAULT_SPEED,
+    animationBounciness: DEFAULT_BOUNCINESS,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      deltaX: 0,
+      deltaY: 0,
+      shouldAnimate: false,
+    };
+    this.ref = React.createRef();
+  }
+
+  componentDidUpdate(prevProps) {
+    const {shouldAnimate} = this.state;
+    const {isPanning, dragDeltas, swipeDirections} = this.props.context;
+    const {
+      isPanning: prevIsPanning,
+      dragDeltas: prevDragDeltas,
+      swipeDirections: prevSwipeDirections,
+    } = prevProps.context;
+    if (isPanning !== prevIsPanning) {
+      if (isPanning && !shouldAnimate) {
+        // do not start a new pan if we're still animating
+        this.onPanStart();
+      } else {
+        this.onPanEnd();
+      }
+    }
+
+    if (
+      isPanning &&
+      (dragDeltas[0] || dragDeltas[1]) &&
```

Comment:
I noticed you're using an array to keep a point data, why not using an object of {x, y} coordinates. 
I think it will be more readable.

Reply:
Done